streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java

```diff
@@ -360,6 +361,115 @@ public void shouldPutAll() {
                 rocksDBStore.get(new Bytes(stringSerializer.serialize(null, "3")))));
     }
 
+    @Test
+    public void shouldReturnKeysWithGivenPrefix() {
+        final List<KeyValue<Bytes, byte[]>> entries = new ArrayList<>();
+        entries.add(new KeyValue<>(
+            new Bytes(stringSerializer.serialize(null, "k1")),
+            stringSerializer.serialize(null, "a")));
+        entries.add(new KeyValue<>(
+            new Bytes(stringSerializer.serialize(null, "prefix_3")),
+            stringSerializer.serialize(null, "b")));
+        entries.add(new KeyValue<>(
+            new Bytes(stringSerializer.serialize(null, "k2")),
+            stringSerializer.serialize(null, "c")));
+        entries.add(new KeyValue<>(
+            new Bytes(stringSerializer.serialize(null, "prefix_2")),
+            stringSerializer.serialize(null, "d")));
+        entries.add(new KeyValue<>(
+            new Bytes(stringSerializer.serialize(null, "k3")),
+            stringSerializer.serialize(null, "e")));
+        entries.add(new KeyValue<>(
+            new Bytes(stringSerializer.serialize(null, "prefix_1")),
+            stringSerializer.serialize(null, "f")));
+
+        rocksDBStore.init((StateStoreContext) context, rocksDBStore);
+        rocksDBStore.putAll(entries);
+        rocksDBStore.flush();
+
+        final KeyValueIterator<Bytes, byte[]> keysWithPrefix = rocksDBStore.prefixScan("prefix", stringSerializer);
+        final List<String> valuesWithPrefix = new ArrayList<>();
+        int numberOfKeysReturned = 0;
+
+        while (keysWithPrefix.hasNext()) {
+            final KeyValue<Bytes, byte[]> next = keysWithPrefix.next();
+            valuesWithPrefix.add(new String(next.value));
+            numberOfKeysReturned++;
+        }
+        // Since there are 3 keys prefixed with prefix, the count should be 3
+        assertThat(numberOfKeysReturned, is(3));
+        // The order might seem inverted to the order in which keys were inserted, but since Rocksdb stores keys
+        // lexicographically, prefix_1 would still be the first key that is returned.
+        assertThat(valuesWithPrefix.get(0), is("f"));
+        assertThat(valuesWithPrefix.get(1), is("d"));
+        assertThat(valuesWithPrefix.get(2), is("b"));
+
+        // Lastly, simple key value lookups should still work :)
+        assertThat("c", is(stringDeserializer.deserialize(null,
+            rocksDBStore.get(new Bytes(stringSerializer.serialize(null, "k2"))))));
```

Comment:
Do we need this verification? This is covered by other unit test methods, isn't it? Could you please remove it from here?